File: flang/test/Driver/print-supported-cpus-aarch64.f90
! Test --print-supported-cpus and associated aliases, -mcpu=help and
! -mtune=help on AArch64.

! REQUIRES: aarch64-registered-target

! RUN: %flang --target=aarch64-unknown-linux-gnu --print-supported-cpus 2>&1 | \
! RUN:   FileCheck %s
! RUN: %flang --target=aarch64-unknown-linux-gnu -mcpu=help 2>&1 | \
! RUN:   FileCheck %s
! RUN: %flang --target=aarch64-unknown-linux-gnu -mtune=help 2>&1 | \
! RUN:   FileCheck %s

! CHECK-NOT: warning: argument unused during compilation

! CHECK: Target: aarch64-unknown-linux-gnu
! CHECK: cortex-a73
! CHECK: cortex-a75

! TODO: This is a line that is printed at the end of the output. The full line
! also includes an example that references clang. That needs to be fixed and a
! a check added here to make sure that it references flang, not clang.

! CHECK: Use -mcpu or -mtune to specify the target's processor.

File: clang/test/Driver/print-supported-cpus-aarch64.c
// Test that --print-supported-cpus lists supported CPU models, including aliases.

// REQUIRES: aarch64-registered-target

// RUN: %clang --target=arm64-apple-macosx --print-supported-cpus 2>&1 | \
// RUN:   FileCheck %s --check-prefix=CHECK --implicit-check-not=apple-latest

// CHECK: Target: arm64-apple-macosx

// CHECK: apple-a11
// CHECK: apple-a12
// CHECK: apple-a13
// CHECK: apple-a14
// CHECK: apple-a15
// CHECK: apple-a16
// CHECK: apple-a17
// CHECK: apple-a7
// CHECK: apple-a8
// CHECK: apple-a9
// CHECK: apple-m1
// CHECK: apple-m2
// CHECK: apple-m3
// CHECK: apple-m4
// CHECK: apple-s4
// CHECK: apple-s5

// CHECK: Use -mcpu or -mtune to specify the target's processor.
